### ISSUE TITLE
/var/log/journal/remote journals not being merged

### ISSUE THREAD

**Issue by kaihendry** [systemd/systemd#3618]
### Submission type
- [x] Bug report
### systemd version the issue has been seen with

```
systemd 230
+PAM -AUDIT -SELINUX -IMA -APPARMOR +SMACK -SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD +IDN
```
### Used distribution

Archlinux
### In case of bug report: Expected behaviour you didn't see

Journals merged from /var/log/journal/remote
### In case of bug report: Unexpected behaviour you saw

Only could see local hostname
### In case of bug report: Steps to reproduce the problem

journalctl -m doesn't include entries in /var/log/journal/remote

If you are feeling very brave the issue comes up in https://www.youtube.com/watch?v=45CQ0tgXQmY

**Comment by tylerjl** [comment]
I would add that I've observed this in version 231 as well (Arch Linux). Although `journalctl` can be pointed at individual journal files received remotely with `-D` or `--file`, I would expect `--merge` to pull all journal files together to be viewed interleaved together (as the documentation mentions).

**Comment by wraithm** [comment]
I've definitely experienced this issue as well.

**Comment by poettering** [comment]
Fixed by #4970
